graylog2-server/src/main/java/org/graylog2/web/customization/Config.java

```diff
@@ -32,7 +32,8 @@ public record Config(
         Optional<Welcome> welcome,
         Optional<Navigation> navigation,
         Optional<Footer> footer,
-        Optional<Resources> resources
+        Optional<Resources> resources,
+        @JsonProperty("investigation_reporting_by_ai") Optional<InvestigationReportingByAI> investigationReportingByAI
```

Comment:
We should not put this at the top level. I would recommend adding a `features` key with an `ai_investigation_report` key, that holds an `{ enabled: boolean }` record. This allows us to add this for other features as well and also allows adding more keys to it besides `enabled`.